.github/workflows/_20_build.yml

```diff
@@ -25,14 +29,25 @@ env:
 
 jobs:
   compile:
-    runs-on: namespace-profile-rust-builder
+    runs-on: namespace-profile-${{ inputs.ci-runner }}
     container:
       image: "nscr.io/n0ie27k9ikobs/chainflip-backend/rust-base:nightly-2025-01-1"
+      volumes:
+        - /cache:/cache
+      options: --cap-add=SYS_ADMIN
+      env:
+        NSC_CACHE_PATH: "/cache"
 
     steps:
       - name: Checkout 🛒
         uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
 
+      - name: Set up Rust cache 🦀💰
+        if: contains(inputs.ci-runner, 'no-cache') != true
```

Comment:
nit: for these conditions I'd do `== false` to avoid the double negative of "no-cache, not equal"